db/rdbms/src/main/resources/mapper/VariableMapper.xml

```diff
@@ -9,30 +9,172 @@
+    resultMap="io.camunda.db.rdbms.sql.VariableMapper.searchResultMap">
+    SELECT
+      VAR_KEY,
+      PROCESS_INSTANCE_KEY,
+      SCOPE_KEY,
+      VAR_NAME,
+      VAR_VALUE,
+    VAR_FULL_VALUE,
+      TENANT_ID,
+    IS_PREVIEW
+    FROM VARIABLE
+    <include refid="io.camunda.db.rdbms.sql.VariableMapper.searchFilter"/>
+    <if test="sort != null and sort.orderings != null and !sort.orderings.isEmpty()">
+      <foreach collection="sort.orderings" open="ORDER BY " separator=", " item="item">
+        <include refid="io.camunda.db.rdbms.sql.VariableMapper.variableSortMapper"/>
+        ${item.order}
+      </foreach>
+    </if>
+    ${paging.after}
+  </select>
+
+  <sql id="searchFilter">
+    WHERE 1 = 1
+    <!-- basic filters -->
+    <if
+      test="filter.processInstanceKeyOperations != null and !filter.processInstanceKeyOperations.isEmpty()">
+      <foreach collection="filter.processInstanceKeyOperations" item="operation">
+        AND PROCESS_INSTANCE_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.scopeKeyOperations != null and !filter.scopeKeyOperations.isEmpty()">
+      <foreach collection="filter.scopeKeyOperations" item="operation">
+        AND SCOPE_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.variableKeyOperations != null and !filter.variableKeyOperations.isEmpty()">
+      <foreach collection="filter.variableKeyOperations" item="operation">
+        AND VAR_KEY
+        <include refid="io.camunda.db.rdbms.sql.Commons.operationCondition"/>
+      </foreach>
+    </if>
+    <if test="filter.tenantIds != null and !filter.tenantIds.isEmpty()">
+      AND TENANT_ID IN
+      <foreach collection="filter.tenantIds" item="value" open="(" separator=", " close=")">#{value}</foreach>
+    </if>
+
+    <!-- advanced filters on variable name / values -->
+    <if test="filter.variableOperations != null and !filter.variableOperations.isEmpty()">
+      AND (
+      <foreach collection="filter.variableOperations" index="name" item="operations">
+        (VAR_NAME = #{name}
+        <foreach collection="operations" item="operation">
+          AND
+          <include refid="variableOperationCondition"/>
+        </foreach>
+        )
+        OR
+      </foreach>
+      false )
+    </if>
+
+  </sql>
+
+  <sql id="variableOperationCondition">
+    <choose>
+      <when test="operation.operator.name().equals('EQUALS')">
+        VAR_VALUE = #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('NOT_EQUALS')">
+        VAR_VALUE != #{operation.value}
+      </when>
+      <when test="operation.operator.name().equals('EXISTS')">
+        VAR_VALUE IS NOT NULL
+      </when>
+      <when test="operation.operator.name().equals('NOT_EXISTS')">
+        VAR_VALUE IS NULL
+      </when>
```

Comment:
❓ The semantic of `exists` and `not_exists` is not clear when searching for variables. Probably, they are not needed and can be removed for now.

Reply:
For search queries where we just search for the name "EXISTS" is used. So it makes no sense here, but we still "get" this operation. I now simply ignore ist.